editor-components/contacts/form3/form3.tsx

```diff
@@ -3,207 +3,836 @@ import { BaseContacts } from "../../EditorComponent";
+      const name = toObjectKey(`input_${indexOfLabel + indexOfInput}`);
+
+      return toObjectKey(name);
+
+    }
+
+    function getInitialValue() {
+      let value: any = {};
+      inputItems.map((inputItem: any, indexOfItem: number) => {
+        inputItem.getPropValue("inputs")?.map((input: TypeUsableComponentProps, indexOfInput: number) => {
+          const key = getInputValue(indexOfItem, input.getPropValue("label", { as_string: true }), indexOfInput);
+          value[key] = "";
+        })
+      })
+      return value;
+    }
+
+    function getSchema() {
+      let schema = Yup.object().shape({});
+
+      inputItems.map((inputItem: any, indexOfItem: number) => {
+        inputItem.getPropValue("inputs").map((input: any, indexOfInput: number) => {
+
+          const key = getInputValue(indexOfItem, input.getPropValue("label", { as_string: true }), indexOfInput);
+
+
+          const isRequired = input.getPropValue("is_required");
+          const isEmail = getInputType(input.getPropValue("type")) == "email";
+
+          let fieldSchema = Yup.string() as any;
+
+          if (isRequired) {
+            fieldSchema = fieldSchema.required(input.getPropValue("required_error_message"))
+          } else {
+            fieldSchema = fieldSchema.nullable();
+          }
+
+          if (isEmail) {
+            fieldSchema = fieldSchema.email(input.getPropValue("type_error_message"))
+          }
+
+          schema = schema.shape({
+            [key]: fieldSchema,
+          });
+        })
+      })
+
+      return schema;
+    };
+
+    function getFormData(obj: any) {
+      const newObj: any = {};
+      Object.values(obj).forEach((value, index) => {
+        newObj["input_" + index] = value;
+      })
+      return newObj;
+    }
+
+    function isRequiredInput(inputItem: any): boolean {
+      return inputItem.getPropValue("inputs").some((input: any) => input.getPropValue("is_required"))
+    }
+
+    function updateLineNumbers(e: React.ChangeEvent<HTMLTextAreaElement>): void {
+      const textarea = e.target;
+      const lineNumbersContainer = document.querySelector('.line-numbers') as HTMLElement;
+      const lines = textarea.value.split('\n').length;
+
+      if (lineNumbersContainer) {
+        lineNumbersContainer.innerHTML = Array.from({ length: lines }, (_, i) => i + 1).join('<br>');
+      }
+    }
+
+    function updateTextareaHeight(el: HTMLTextAreaElement): void {
+      if (el) {
+        el.style.height = '20px';
+        const scrollHeight = el.scrollHeight;
+        if (scrollHeight < 50) {
+          el.style.height = scrollHeight + 'px';
+        } else {
+          el.style.height = '50px';
+          el.style.overflowY = 'scroll';
+        }
+      }
+    }
+    document.addEventListener('DOMContentLoaded', () => {
+      const textarea = document.querySelector('.input') as HTMLTextAreaElement;
+      updateLineNumbers({ target: textarea } as React.ChangeEvent<HTMLTextAreaElement>);
+    });
```

Comment:
that querySelector not work cause we are using module CSS. so the classNames will come different (ex: input returns input__NSskWASD). check this code if it not work remove it. also delete updateLineNumbers method